[22788] Documentation thought doxygen and readthedocs && README (#10)
eProsima/FIWARE-DDS-Enabler: README.md

  <a href="https://github.com/eProsima/FIWARE-DDS-Enabler/actions/workflows/nightly-ubuntu-ci.yml"><img alt="Ubuntu CI" src="https://img.shields.io/github/actions/workflow/status/eProsima/FIWARE-DDS-Enabler/nightly-ubuntu-ci.yml?label=Ubuntu%20CI"></a>
</div>

<br><br>

*eProsima DDS Enabler* is a modular middleware solution that bridges DDS networks and FIWARE NGSI-LD Context Brokers, providing real-time, bidirectional interoperability. It manages all required DDS participants, auto-discovers topics and types, and seamlessly routes DDS samples into NGSI-LD entities and context updates back into DDS topics.

**Key features**
- **Unified DDS Participant Management**
  Auto-create and discover DomainParticipants, Publishers, Subscribers, Topics and Types without manual code.
- **Flexible YAML Configuration**
  Fine-tune QoS, network filters, topic allow-listing/deny-listing and discovery via a human-readable YAML file.
- **Dynamic Types via XTypes**

>   Leverage `OMG DDS-XTypes 1.3 <https://www.omg.org/spec/DDS-XTypes/1.3>`_ and Fast DDS serialization utilities for runtime type registration and discovery.

Reviewer: Make this a markdown link

- **Core Engine Powered by DDS-Pipe**

>   Built on `eProsima DDS Pipe <https://github.com/eProsima/DDS-Pipe>`_, ensuring low-latency, high-throughput payload forwarding and reliable discovery across distributed systems.

Reviewer: Make this a markdown link

- **Serialization Utilities**

>   Convert DDS data to JSON for REST integration and to human-readable IDL; JSON→DDS deserializer support is in development.

Reviewer: ```suggestion
  Convert DDS data to JSON and vice versa for REST integration and to human-readable IDL.
```

Reviewer: Apply this to the documentation overview section

- **Bidirectional FIWARE Integration**
  Routes DDS samples to NGSI-LD Context Brokers and injects context updates back into DDS topics.

![DDS Enabler Architecture](docs/rst/figures/ddsenabler_architecture.png)

> ## Commercial support
> 
> Looking for professional support or custom integration?
> Contact us at **info@eprosima.com** or raise an issue on GitHub.

Reviewer: ```suggestion
## Commercial support

Looking for commercial support? Write us to info@eprosima.com

Find more about us at [eProsima’s webpage](https://eprosima.com/).
```

## Documentation

> Explore the full user and developer guide hosted on Read the Docs:

Reviewer: ```suggestion
Explore the full user and developer guide hosted on ReadTheDocs:
```
